Shared/Samples/Show device location/ShowDeviceLocationView.swift

```diff
@@ -17,15 +17,32 @@ import CoreLocation
 import SwiftUI
 
 struct ShowDeviceLocationView: View {
+    /// A Boolean value indicating whether to show an alert.
+    @State private var isShowingAlert = false
+    
+    /// The error shown in the alert.
+    @State private var error: Error? {
+        didSet { isShowingAlert = error != nil }
+    }
+    
+    /// A Boolean value indicating whether the settings button is disabled.
+    @State private var areSettingsDisabled = true
+    
     /// The view model for this sample.
     @StateObject private var model = Model()
     
     var body: some View {
         MapView(map: model.map)
             .locationDisplay(model.locationDisplay)
             .task {
-                await model.startLocationDataSource()
-                await model.updateAutoPanMode()
```

Comment:
This was a trivial bug in this sample. Before this change the task modifier will get stuck, waiting for the auto pan mode stream to end indefinitely. Now it is wrapped in a Task block and has the same lifespan as the model object.

Reply:
I made some changes here, please take another look in the next round of review.